tests/cpydiff/core_class_name_mangling: New tests for name mangling.

trwbox opened the pull request:
Adding new tests/documentation for missing name mangling for private class members.

Adds testing/documentation for #12150.

trwbox committed:
tests/cpydiff: Add new CPy diff test for class name mangling.

Adds new tests/documentation for missing name mangling for private class
members.

Signed-off-by: Trent Warlaven <trwbox@gmail.com>

dlech commented on tests/cpydiff/core_class_name_mangling.py:
```diff
@@ -0,0 +1,25 @@
+"""
+categories: Core,Classes
+description: Private Class Members name mangling is not implemented
+cause: THe MicroPython compiler has not yet implemented name mangling for private class members at runtime.
```
Typo and saying "not yet" makes it sound like there are plans to implement it, which is not currently true.

```suggestion
cause: The MicroPython compiler does not implement name mangling for private class members at runtime.
```

dlech replied:
Git blame says the comment is from the initial commit of MicroPython 10 years ago, so likely doesn't reflect the current state of things. I guess Damien can chime in here if he wants, but it seems like if it hasn't been implemented by now, we don't really need it.

dpgeorge replied:
The vast majority of tests in `tests/cpydiff` use language like "not implemented", so I suggest to also use that language here, ie remove the "yet".

Then I think it's a good idea to remote that TODO in `compile.c`.  This test will encode the intention of that to-do.

dpgeorge replied:
I updated the wording as suggested, and removed the TODO in `compile.c`.

dlech commented on tests/cpydiff/core_class_name_mangling.py:
```diff
@@ -0,0 +1,25 @@
+    def do_print(self):
+        __print_string(self.string)
+
+example_string = "Example String to print."
+print(example_string)
+
+class_item = Foo(example_string)
+print(class_item.string)
+
+class_item.do_print()
```
Do we need these first two `print()`s since they are not demonstrating the difference?

```suggestion
class_item = Foo("Example String to print.")
class_item.do_print()
```

trwbox replied:
The first print can definitely be removed. I added the `print(class_item.string)` to show the attribute for the instance is usable, and not part of the issue. It isn't needed, but to me it provides clarity, when reading the generated documentation, if you don't already know what name mangling is? 

Since it is just showing it is a usable property, is there a better way to show that, that doesn't involve printing it, that would make that idea more clear, if it should be kept at all?

dlech replied:
> It isn't needed, but to me it provides clarity, when reading the generated documentation, if you don't already know what name mangling is?

I already know what name mangling is, so I'm probably not the best person to ask. :smile: 

> is there a better way to show that, that doesn't involve printing it, that would make that idea more clear, if it should be kept at all?


I think using `print()` is a good way to show the "did not raise exception" case and is consistent with other CPy diff snippets.

dpgeorge replied:
I think just remove the first `print`.

dpgeorge replied:
I removed the first print.

dlech reviewed (commented):
Thanks for the PR.

Could you run `black` on the new file (or manually fix the code format issues as indicated in the CI test failure)?

dpgeorge committed:
py/compile: Remove TODO about name mangling.

This TODO is now covered by the tests/cpydiff/core_class_name_mangling.py
test.

Signed-off-by: Damien George <damien@micropython.org>
